src/_pytest/subtests.py

```diff
@@ -0,0 +1,431 @@
+    suspend_capture_ctx: Callable[[], AbstractContextManager[None]]
+    request: SubRequest
+    reporter: SubtestsReporterPlugin
+
+    def __enter__(self) -> None:
+        __tracebackhide__ = True
+
+        self._start = time.time()
+        self._precise_start = time.perf_counter()
+        self._exc_info = None
+
+        self._exit_stack = ExitStack()
+        self._captured_output = self._exit_stack.enter_context(
+            capturing_output(self.request)
+        )
+        self._captured_logs = self._exit_stack.enter_context(
+            capturing_logs(self.request)
+        )
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> bool:
+        __tracebackhide__ = True
+        try:
+            if exc_val is not None:
+                exc_info = ExceptionInfo.from_exception(exc_val)
+            else:
+                exc_info = None
+        finally:
+            self._exit_stack.close()
+
+        precise_stop = time.perf_counter()
+        duration = precise_stop - self._precise_start
+        stop = time.time()
+
+        call_info = CallInfo[None](
+            None,
+            exc_info,
+            start=self._start,
+            stop=stop,
+            duration=duration,
+            when="call",
+            _ispytest=True,
+        )
+        report = self.ihook.pytest_runtest_makereport(
+            item=self.request.node, call=call_info
+        )
+        sub_report = SubtestReport._from_test_report(
+            report, SubtestContext(msg=self.msg, kwargs=self.kwargs.copy())
+        )
+
+        if sub_report.failed:
+            self.reporter.contains_failed_subtests[self.request.node.nodeid] += 1
+
+        self._captured_output.update_report(sub_report)
+        self._captured_logs.update_report(sub_report)
+
+        with self.suspend_capture_ctx():
+            self.ihook.pytest_runtest_logreport(report=sub_report)
+
+        if check_interactive_exception(call_info, sub_report):
+            self.ihook.pytest_exception_interact(
+                node=self.request.node, call=call_info, report=sub_report
+            )
+
+        if exc_val is not None:
+            if self.request.session.shouldfail:
+                return False
+        return True
+
+
+@contextmanager
+def capturing_output(request: SubRequest) -> Iterator[Captured]:
+    option = request.config.getoption("capture", None)
+
+    # capsys or capfd are active, subtest should not capture.
+    capman = request.config.pluginmanager.getplugin("capturemanager")
+    capture_fixture_active = getattr(capman, "_capture_fixture", None)
+
+    if option == "sys" and not capture_fixture_active:
+        fixture = CaptureFixture(SysCapture, request, _ispytest=True)
+    elif option == "fd" and not capture_fixture_active:
+        fixture = CaptureFixture(FDCapture, request, _ispytest=True)
+    else:
+        fixture = None
```

Comment:
To me the code is clearer written like this:

```suggestion
    capman = request.config.pluginmanager.getplugin("capturemanager")
    if getattr(capman, "_capture_fixture", None):
        # capsys or capfd are active, subtest should not capture.
        fixture = None
    elif option == "sys":
        fixture = CaptureFixture(SysCapture, request, _ispytest=True)
    elif option == "fd":
        fixture = CaptureFixture(FDCapture, request, _ispytest=True)
    else:
        fixture = None
```